src/Management/Console/Command/GenerateEntryClassesCommand.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Management\Console\Command;
+
+use Contentful\Management\Client;
+use Contentful\Management\CodeGenerator\Entry;
+use Contentful\Management\CodeGenerator\Loader;
+use Contentful\Management\CodeGenerator\Mapper;
+use Contentful\Management\Query;
+use Contentful\Management\Resource\ContentType;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * GenerateEntryClassesCommand class.
+ *
+ * CLI command for generating entry classes based off content types.
+ */
+class GenerateEntryClassesCommand extends Command
+{
+    /**
+     * @var array
+     */
+    private $createdFiles = [];
+
+    /**
+     * @var string
+     */
+    private $loaderFile;
+
+    protected function configure()
+    {
+        $this
+            ->setName('generate:entry-classes')
+            ->setDescription('Generates entry classes based off available content types')
+            ->setDefinition([
+                new InputArgument(
+                    'space-id',
+                    InputArgument::REQUIRED,
+                    'ID of the space to use.'
+                ),
+                new InputArgument(
+                    'token',
+                    InputArgument::REQUIRED,
+                    'Token to access the space.'
+                ),
+                new InputArgument(
+                    'dir',
+                    InputArgument::REQUIRED,
+                    'The directory to write the cache to.'
+                ),
+                new InputArgument(
+                    'namespace',
+                    InputArgument::OPTIONAL,
+                    'The base namespace to use.',
+                    ''
+                ),
+            ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $spaceId = $input->getArgument('space-id');
+        $dir = $input->getArgument('dir');
+
+        $this->performDirectoryChecks($dir);
+
+        $this->generate(
+            $input->getArgument('token'),
+            $spaceId,
+            $input->getArgument('namespace')
+        );
+
+        $this->writeFiles($dir);
+
+        $this->writeReport($output, $spaceId, $dir);
+    }
+
+    /**
+     * @param string $dir
+     *
+     * @throws \RuntimeException
+     */
+    private function performDirectoryChecks(string $dir)
+    {
+        // @TODO
```

Comment:
Hmm, crazy idea: Should we create a tp ticket and add the ticket id here? Or even have a github issue to track todos?

Reply:
I'm working on this now, I just wanted to submit the bulk so you guys could already take a look 🙂